From bacd3a44fe6f06ddd8bb65a3a56eb1738a65ed8c Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Fri, 6 Mar 2020 16:00:23 +0100
Subject: [PATCH] Don't build/install fwupdagent man page if agent build is not
 requested

Otherwise build fails with:

  src/meson.build:196:2: ERROR: Unknown variable "fwupdagent".

Gentoo-bug: https://bugs.gentoo.org/711682
Signed-off-by: Lars Wendler <polynomial-c@gentoo.org>
---
 src/meson.build | 30 ++++++++++++++++--------------
 1 file changed, 16 insertions(+), 14 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index 46a9c4ded..e9aac607e 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -193,20 +193,22 @@ if build_daemon and get_option('man')
     install : true,
     install_dir : join_paths(mandir, 'man1'),
   )
-  custom_target('fwupdagent-man',
-    input : fwupdagent,
-    output : 'fwupdagent.1',
-    command : [
-      help2man, '@INPUT@',
-      '--no-info',
-      '--output', '@OUTPUT@',
-      '--name', 'Firmware updating agent',
-      '--manual', 'User Commands',
-      '--version-string', fwupd_version,
-    ],
-    install : true,
-    install_dir : join_paths(mandir, 'man1'),
-  )
+  if get_option('agent')
+    custom_target('fwupdagent-man',
+      input : fwupdagent,
+      output : 'fwupdagent.1',
+      command : [
+        help2man, '@INPUT@',
+        '--no-info',
+        '--output', '@OUTPUT@',
+        '--name', 'Firmware updating agent',
+        '--manual', 'User Commands',
+        '--version-string', fwupd_version,
+      ],
+      install : true,
+      install_dir : join_paths(mandir, 'man1'),
+    )
+  endif
 endif
 if get_option('man')
   custom_target('fwupdtool-man',
